deflate_medium.c

```diff
@@ -189,14 +190,15 @@ static void fizzle_matches(deflate_state *s, struct match *current, struct match
 }
 
 ZLIB_INTERNAL block_state deflate_medium(deflate_state *s, int flush) {
-    struct match current_match, next_match;
+    ALIGNED_(16) struct match current_match;
+    struct match next_match;
```

Comment:
This doesn't look right... Either neither should be aligned or both should be aligned.

Reply:
Looks good to me

Reply:
The ```struct match``` is 8 bytes (4x2 bytes), and aligning it on 16 byte boundary is overkill.

Reply:
I aligned it on a 16-byte boundary to force it to start at a new cacheline, since it is 8 bytes it only takes half of the cacheline, leaving room for the next struct to take up the second half of the cacheline. Thus both structs live in a single cacheline.
This is how I have seen this done other places, but I understand it can look a bit confusing, so maybe a comment would help?

Reply:
It would be a lot clearer if it would be nested struct instead of two separate structs... But it definitely needs a comment.

Reply:
Comment added. Also indented the second struct to make it easier to read and to see that they are related.